zilencer/models.py

```diff
@@ -63,10 +63,14 @@ class RemotePushDeviceToken(AbstractPushDeviceToken):
 
     server: RemoteZulipServer = models.ForeignKey(RemoteZulipServer, on_delete=models.CASCADE)
     # The user id on the remote server for this device
-    user_id: int = models.BigIntegerField(db_index=True)
+    user_id: int = models.BigIntegerField(db_index=True, null=True)
+    user_uuid: UUID = models.UUIDField(null=True)
 
     class Meta:
-        unique_together = ("server", "user_id", "kind", "token")
+        unique_together = [
+            ("server", "user_id", "kind", "token"),
+            ("server", "user_uuid", "kind", "token"),
+        ]
```

Comment:
I guess this is implicitly taking advantage of the fact that Django unique indexes will ignore cases where one of the 4 fields, in this case `user_uuid`, is `null`?  I think that's OK, but would probably benefit from a comment explaining that detail.

Reply:
I think this is a feature of Postgres, not of Django?

Reply:
Yeah, I think that's right; but the point is that this index doesn't do what it appears to do to the naive reader with respect to nulls (namely allowing only one (server, null, kind, token) row to exist), and that's what we want, but we should point out that we're taking advantage of this subtle behavior.

Reply:
Added a comment on that here - however, `token` is generally  going be unique on its own I think? So that alone will cause that the `null` aspect won't actually come into play.